### PR TITLE
fix: Expose jaeger knobs and default max-packet-size to something that works everywhere

### PR DESCRIPTION
**Pull request by mkmik**
`--traces-exporter-jaeger-max-packet-size` is important also when you run the jaeger collector
on "localhost" by running `docker run jaegertracing/all-in-one ....` which on mac doesn't really
work on the real localhost but has a few hops between tunneling interfaces, so you'd get mysteriously
dropped packets that can easily drive you to doubt your own sanity on an otherwise calm Thursday evening.

### CODE REVIEW

File: src/commands/tracing.rs [hunk 1]
```diff
@@ -160,13 +160,25 @@ fn construct_opentelemetry_tracer(config: &crate::commands::run::Config) -> Opti
...
+                    .with_service_name(&config.traces_exporter_jaeger_service_name)
+                    .with_max_packet_size(config.traces_exporter_jaeger_max_packet_size);
+
+                // Batching is hard to tune because the max batch size
+                // is not currently exposed as a tunable from the trace config, and even then
+                // it's defined in terms of max number of spans, and not their size in bytes.
+                // Thus we enable batching only when the MTU size is 65000 which is the value suggested
+                // by jaeger when exporting to localhost.
+                if config.traces_exporter_jaeger_max_packet_size >= 65_000 {
+                    builder.install_batch(opentelemetry::runtime::Tokio)
```
Comment (alamb) [inline src/commands/tracing.rs:176]: I have no idea what implications this has, but if @jacobmarble  is cool with it so am I 👍

**Review by alamb — APPROVED** [review]
code looks to do what it intends

File: src/commands/tracing.rs (hunk 1 above)
Comment (mkmik) [reply src/commands/tracing.rs:176]: The batching feature is brittle. Even with 65000 bytes, you get this error when you log a lot:

```
OpenTelemetry trace error occurred ExportFailed(ThriftAgentError(ProtocolError { kind: SizeLimit, message: "jaeger exporter payload size of 68898 bytes over max UDP packet size of 65000 bytes. Try setting a smaller batch size." }))
```

And that's when everything works fine.

When the collector just pretends to be on localhost but instead is just the big fat illusion called docker for mac, you can get these very insightful errors:

```
OpenTelemetry trace error occurred ExportFailed(ThriftAgentError(TransportError { kind: Unknown, message: "other os error" }))
```

My suggestion here is to have safe defaults, and let people tune these parameters to unsafe ranges when they know what they are doing, when they deploy the code.

File: src/commands/run.rs [hunk 2]
```diff
@@ -245,6 +245,32 @@ pub struct Config {
...
+    #[structopt(
+        long = "--traces-exporter-jaeger-service-name",
+        env = "TRACES_EXPORTER_JAEGER_SERVICE_NAME",
+        default_value = "iox"
+    )]
+    pub traces_exporter_jaeger_service_name: String,
+
+    /// Tracing: Jaeger max UDP packet size
+    ///
+    /// Default to 1500, which is the default ethernet MTU.
```
Comment (jacobmarble) [inline src/commands/run.rs:260]: 1500 bytes is often MTU (ethernet standard), but:
- UDP overhead is 8 bytes
- network links like PPPoE steal more than that
- Wireguard MTU is 1420 by default

This should probably be considered a bug in the OpenTelemetry Rust SDK. Handling it properly could lead to madness. We haven't run into this problem (that I'm aware of) in Go.

**Comment by jacobmarble** [comment]
This issue is being tracked already
https://github.com/open-telemetry/opentelemetry-rust/issues/545

File: src/commands/run.rs (hunk 2 above)
Comment (mkmik) [reply src/commands/run.rs:260]: the size of the header is often included in the calculation of the "packet size"; but I cannot vouch for this library implementation.  Good point on the other common causes for reduced MTU. Probably 1300 is a decent default (e.g. AWS wavelength has that MTU)

**Pull request merged**